app/models/manageiq/providers/ibm_cloud/inventory/parser/vpc.rb

```diff
@@ -40,6 +45,8 @@ def instances
         :location          => instance&.dig(:zone, :name) || "unknown",
         :genealogy_parent  => persister.miq_templates.lazy_find(instance&.dig(:image, :id)),
         :availability_zone => persister.availability_zones.lazy_find(instance&.dig(:zone, :name)),
+        :flavor            => persister.flavors.lazy_find(instance&.dig(:profile, :name)),
+        :key_pairs         => instance_key_pairs(instance[:id]).compact,
```

Comment:
I don't think the `.compact` is necessary because you're not e.g. using `next if something` in the block which would lead to `nil`s in your array.